src/randomeffectsterm.jl

```diff
@@ -27,8 +27,14 @@ function StatsModels.apply_schema(
     if length(t.args_parsed) ≠ 2
         throw(ArgumentError("malformed nesting term: $t (Exactly two arguments required"))
     end
-    first, second = apply_schema.(t.args_parsed, Ref(sch.schema), Mod)
-    return first + first & second
+
+    first, second = apply_schema.(t.args_parsed, Ref(sch), Mod)
+    
+    if !(typeof(first) <: CategoricalTerm)
+        throw(ArgumentError("nesting terms requires categorical grouping term, got $first.  Manually specify $first as `CategoricalTerm` in hints/contrasts"))
+    end
```

Comment:
I think this error is the only thing not covered by the tests. Can you add one?

Reply:
Yup.  And one for nesting too deep, too (in case anyone gets any ideas ;) )